packages/flutter/lib/src/material/tab_indicator.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+
+/// Used with [TabBar.indicatorShape] to draw a horizontal line below the
+/// selected tab
+///
+/// The line is inset from the tab's boundary by [padding]. The [borderSide]
+/// defines the line's color and weight.
+///
+/// The [TabBar.indicatorSize] property can be used to define the
+/// tab's boundary in terms of its (centered) widget, [TabIndicatorSize.label],
+/// or the the entire tab, [TabIndicatorSize.tab].
+class DefaultTabIndicator extends ShapeBorder {
+  /// Create a underline style tab indicator.
+  ///
+  /// The [borderSide] and [padding] arguments must not be null.
+  const DefaultTabIndicator({
+    this.borderSide: const BorderSide(width: 2.0, color: Colors.white),
+    this.padding: EdgeInsets.zero,
+  }) : assert(borderSide != null), assert(padding != null);
+
+  /// The color and weight of the line drawn below the selected tab.
```

Comment:
here and elsewhere in this class: consider reframing the documentation to describe this just as a shape, rather than as an indicator-specific feature.